sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsLoader.java

```diff
@@ -0,0 +1,151 @@
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.xml.sax.Attributes;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractLoader;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Loads a filter chain of suppressions.
+ *
+ * @author Rick Giles
+ */
+public final class ForbidCertainMethodsLoader
+        extends AbstractLoader {
+    /**
+     * The public ID for the configuration dtd.
+     */
+    private static final String DTD_PUBLIC_ID_1_0 =
+            "-//Puppy Crawl//DTD Forbidden Methods 1.0//EN";
+    /**
+     * The resource for the configuration dtd.
+     */
+    private static final String DTD_RESOURCE_NAME_1_0 =
+            "com/puppycrawl/tools/checkstyle/forbidden_methods_1_0.dtd";
+    /**
+     * File search error message.
+     **/
+    private static final String UNABLE_TO_FIND_ERROR_MESSAGE = "Unable to find: ";
+
+    /**
+     * The filter chain to return in getAFilterChain(),
+     * configured during parsing.
+     */
+    private final List<ForbidCertainMethodsCheck.CallInfo> callInfoList = new ArrayList<>();
+
+    /**
+     * Creates a new {@code SuppressionsLoader} instance.
+     *
+     * @throws ParserConfigurationException if an error occurs
+     * @throws SAXException                 if an error occurs
+     */
+    private ForbidCertainMethodsLoader()
+            throws ParserConfigurationException, SAXException {
+        super(Collections.singletonMap(DTD_PUBLIC_ID_1_0, DTD_RESOURCE_NAME_1_0));
+    }
+
+    @Override
+    public void startElement(String namespaceUri,
+                             String localName,
+                             String qName,
+                             Attributes attributes)
+            throws SAXException {
+        if ("ForbiddenMethod".equals(qName)) {
+            //add SuppressElement filter to the filter chain
+            final String methodName = attributes.getValue("methodName");
+            final String argCount = attributes.getValue("argCount");
+            if (methodName == null) {
+                // -@cs[IllegalInstantiation] SAXException is in the overridden method signature
+                throw new SAXException("missing methodName attribute");
+            }
+            if (argCount == null) {
+                // -@cs[IllegalInstantiation] SAXException is in the overridden method signature
+                throw new SAXException("missing argCount attribute");
+            }
+            // ensure that regex is valid
+            Pattern.compile(methodName);
+            Pattern.compile(argCount);
+            callInfoList.add(ForbidCertainMethodsCheck.CallInfo.of(methodName, argCount));
```

Comment:
We should be sending the compiled patterns and not the Strings.
`matches` will just run the compile again and each time it looks creating a performance hit.

Reply:
I will make the change.